--- lwnfs.c 2009-02-09 11:16:04.000000000 -0600
+++ lwnfs.c 2009-02-09 11:29:42.000000000 -0600
@@ -9,6 +9,12 @@
* nethack high scores, and set your disk drives on fire. You have
* been warned.
*/
+
+/*
+ * Return type of lfs_get_super should be int. 
+ * lfs_get_super and get_sb_single need a parameter of type struct vfsmount *
+ * Chris Brannon, 02/09/2009
+*/
#include <linux/kernel.h>
#include <linux/init.h>
#include <linux/module.h>
@@ -154,10 +160,11 @@
/*
* Stuff to pass in when registering the filesystem.
*/
-static struct super_block *lfs_get_super(struct file_system_type *fst,
-  int flags, const char *devname, void *data)
+static int lfs_get_super(struct file_system_type *fst,
+  int flags, const char *devname, void *data,
+  struct vfsmount *mount)
{
- return get_sb_single(fst, flags, data, lfs_fill_super);
+ return get_sb_single(fst, flags, data, lfs_fill_super, mount);
}

static struct file_system_type lfs_type = {
